### ISSUE TITLE
Custom validators for format of strings

### ISSUE THREAD

**Issue by domenkozar** [p1c2u/openapi-core#56]
As per https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#data-types

```
However, to support documentation needs, the format property is an open string-valued property, and can have any value. Formats such as "email", "uuid", and so on, MAY be used even though undefined by this specification. Types that are not accompanied by a format property follow the type definition in the JSON Schema. Tools that do not recognize a specific format MAY default back to the type alone, as if the format is not specified.
```

It would be great if openapi-core allows custom validation of `format: uuid`, etc.

**Comment by diogobaeder** [comment]
Hi guys,

This doesn't seem to have been fixed; I tried defining a custom formatter for the UUID type, like this:

```python
request_validator = RequestValidator(spec, {
    'uuid': Format(UUID, TypeValidator(UUID)),
})
```
But this simply doesn't work, because `UUID` is not listed as one of the valid types which are checked on validation time - here's the code in the library that creates the validator that contains those types: 
https://github.com/p1c2u/openapi-core/blob/master/openapi_core/schema/schemas/models.py#L56

In other words, as soon as the value received is converted to the internal type (`UUID`), then this can't get past validation.

Is this a bug or should I have done something else in order to make it work with UUID-like strings?

Thanks!

**Comment by pbasista** [comment]
I believe that the same is true for `email` format.
